### PR TITLE
Kotlin generation improvements

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Description
- Move `from` into companion object
- Fix nullability of arguments and default value
- Do not attach metadata if it's null
- Log invalid `from` statements
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt [hunk 1]
```diff
@@ -40,21 +44,32 @@ object TelemetryGenerator {
...
-                .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
-                .addParameter("type", Any::class)
-                .addStatement("return values().firstOrNull { it.name == type.toString() }").build()
-        )
-        enum.addKdoc(item.description)
+
+        val companion = TypeSpec.companionObjectBuilder()
+            .addFunction(
+                FunSpec.builder("from")
+                    .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
```
Comment (kiiadi) [inline telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: To me `from` is effectively a constructor so it should never return null. Either you get back an object or it blows up.

File: telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt [hunk 2]
```diff
@@ -40,21 +44,32 @@ object TelemetryGenerator {
...
+                    .addParameter("type", Any::class.asTypeName().copy(nullable = true))
```
Comment (kiiadi) [inline telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: Similarly, why can't this be non-nullable?

File: telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt (hunk 1 above)
Comment (hunterwerlla) [reply telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: That was my initial thought, but I'm hesitant to throw as a result of telemetry. I can easily imagine that we make a change and forget to change the telemetry which would actually affect a workflow

Comment (hunterwerlla) [reply telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: Had a conversation offline, reverted this change and made it able to throw again

File: telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt (hunk 2 above)
Comment (hunterwerlla) [reply telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: back to being non nullable

File: telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt (hunk 1 above)
Comment (kiiadi) [reply telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt]: Not sure what the convo was, and I agree that generally we don't want Telemetry to break the general flow - but this should only be the case if there's a code-problem at build time - right? so it should be OK in this case.

**Pull request merged**